src/measure_theory/measure/finite_measure_weak_convergence.lean

```diff
@@ -0,0 +1,276 @@
+
+## Notations
+
+No new notation is introduced.
+
+## Implementation notes
+
+The topology of weak convergence of finite Borel measures will be defined using a mapping from
+`finite_measures α` to `weak_dual ℝ≥0 (α →ᵇ ℝ≥0)`, inheriting the topology from there.
+
+## References
+
+* [Billingsley, *Convergence of probability measures*][billingsley1999]
+
+## Tags
+
+weak convergence of measures, finite measure, probability measure
+
+-/
+
+noncomputable theory
+open measure_theory
+open set
+open filter
+open_locale topological_space ennreal nnreal
+
+section finite_measures
+/-!
+### Finite measures
+In this section, we define the `Type` of finite measures on a given measurable space. This type
+will be equipped with the topology of weak convergence of measures when the measurable space is
+a topological space equipped with its Borel sigma-algebra.
+-/
+
+variables {α : Type} [measurable_space α]
+
+/-- Finite measures are defined as the subtype of measures that have the property of being finite
+measures (i.e., their total mass is finite). -/
+def finite_measures (α : Type*) [measurable_space α] : Type :=
+{ μ : measure α // finite_measure μ }
+
+namespace finite_measures
+
+instance has_zero {α : Type*} [measurable_space α] :
+  has_zero (finite_measures α) :=
+⟨{ val := 0,
+   property := measure_theory.finite_measure_zero, }⟩
+
+instance : inhabited (finite_measures α) := { default := 0 }
+
+instance {α : Type*} [measurable_space α] : has_add (finite_measures α) :=
+{ add := (λ (μ ν : finite_measures α),
+  { val := μ.val + ν.val,
+    property := @measure_theory.finite_measure_add α _ μ.val ν.val μ.prop ν.prop, }), }
+
+instance {α : Type*} [measurable_space α] : has_scalar ℝ≥0 (finite_measures α) :=
+{ smul := (λ (c : ℝ≥0) (μ : finite_measures α),
+  { val := c • μ.val,
+    property := @measure_theory.finite_measure_smul_nnreal α _ μ.val μ.prop c, }), }
+
+-- TODO: A coercion `finite_measures α → measure α` is crucial. Now it is the subtype coercion.
+-- Another alternative is to use a bijection with `vector_measure α ℝ≥0` as an intermediate step.
+-- Potential advantages of the vector measure alternative:
+--  * Structures of additive commutative monoid and ℝ≥0 module on `finite_measure α` could be
+--    "inherited".
+-- Potential drawbacks of the vector measure alternative:
+--  * No integration theory directly.
+--  * Monotonicity lost as non-measurable sets defined to have measure 0.
+
+/-- A finite measure can be interpreted as a measure. -/
+instance : has_coe (finite_measures α) (measure_theory.measure α) := coe_subtype
+
+instance to_measure.finite_measure (μ : finite_measures α) :
+  finite_measure (μ : measure α) := μ.prop
+
+instance (α : Type*) [measurable_space α] :
+  has_coe_to_fun (finite_measures α) := ⟨(λ _, set α → ℝ≥0),
+    (λ μ, (λ s, (μ.val.measure_of s).to_nnreal))⟩
+
+lemma to_fun_eq_to_measure_to_nnreal (ν : finite_measures α) :
+  (ν : set α → ℝ≥0) = λ s, ((ν : measure α) s).to_nnreal := rfl
```

Comment:
Can you please add another lemma (I think it can be a simp-lemma) saying that, for `s : set α`,
```
(ν s : ℝ≥0∞) = (ν : measure α) s
```
I would actually guess that in 80% of use cases this second lemma is what's needed.

Reply:
That didn't work directly (the RHS is an `ennreal`), but I created one stating
``` 
ν s = ((ν : measure α) s).to_nnreal
```
and I added a prime to the name of the old one (anticipating it is used less often).

Reply:
Sorry for not checking it out myself before I made the comment.  I meant for the LHS to be coerced to ennreal; maybe it needs to be
```
(↑(ν s) : ℝ≥0∞) = (ν : measure α) s
```

Reply:
No, sorry, I had misunderstood your original comment! I am too obsessed about whether the `nnreal`-valuedness will work well in the long term, so I saw what I expected, not what you actually wrote :grimacing:.  Sorry!